azure-pipelines-PR.yml

```diff
@@ -0,0 +1,87 @@
+trigger:
+  - main
+
+pr:
+  - main
+
+variables:
+- template: eng/common-variables.yml
+
+resources:
+  containers:
+  - container: UbuntuContainer
+    image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-cross
+
+stages:
+- stage: build
+  displayName: Build
+  jobs:
+  - template: /eng/common/templates/jobs/jobs.yml
+    parameters:
+      workspace:
+        clean: all
+      jobs:
+      - job: Windows_NT
+        timeoutInMinutes: 90
+        pool:
+          name: NetCore-Public
+          demands: ImageOverride -equals Windows.vs2022.amd64.open
+
+        strategy:
+          matrix:
+              Build_Debug:
+                _BuildConfig: Debug
+        preSteps:
+        - checkout: self
+          clean: true
+        steps:
+        - task: UseDotNet@2
+          displayName: Get .NET Core SDK
+          inputs:
+            version: 8.x
+        - task: DotNetCoreCLI@2
+          displayName: Run Tests
+          inputs: 
+            command: test
+
+  - job: LinuxTests
+    timeoutInMinutes: 30
+    container: UbuntuContainer
+    pool:
+      name: NetCore-Public
+      demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
```

Comment:
this is a pretty old version of Ubuntu - at what point should we update?

Reply:
Follow-up question: _should_ this be running tests on like a min-spec OS?

Reply:
looking at vstest pipelines state, the image selection seems to be reasonable: 
https://github.com/microsoft/vstest/blob/main/azure-pipelines.yml